lib/internal/async_hooks.js

```diff
@@ -332,6 +338,38 @@ function emitDestroyScript(asyncId) {
 }
 
 
+// This is the equivalent of the native push_async_ids() call.
+function pushAsyncIds(asyncId, triggerAsyncId) {
```

Comment:
Curious, why does this function exist? Looks like it's used by JS class AsyncResource only (eventually, via emitBeforeScript), why don't we call the native pushAsyncIds_() in emitBeforeScript() directly if it does exactly the same thing?

Reply:
It is to avoid a C++ call in most cases.